From 844f827363f89331bba60bc97388dc9e2640d65b Mon Sep 17 00:00:00 2001
From: Pali <pali@cpan.org>
Date: Sun, 20 Nov 2016 21:04:27 +0100
Subject: [PATCH] Fix parsing configure libs from mysql_config --libs output
 in Makefile.PL

Libraries in mysql_config --libs output can be specified by library name
with -l prefix or by absolute path to library name without any prefix.
Parameters must start with a hyphen, so treat all options without leading
hyphen in mysql_config --libs output as libraries with full path.
---
 Makefile.PL |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 0e78f00..a2202a3 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -578,7 +578,7 @@ sub Configure {
                 if ($param eq 'libs') {
                     my (@libs, @ldflags);
                     for (split ' ', $str) {
-                        if (/^-[Ll]/) { push @libs, $_ }
+                        if (/^-[Ll]/ || /^[^\-]/) { push @libs, $_ }
                         else          { push @ldflags, $_ }
                     }
                     $str = "@libs";
-- 
1.7.9.5

